numpy/core/arrayprint.py

```diff
@@ -455,7 +495,7 @@ def array2string(a, max_line_width=None, precision=None,
             lst = format_function(arr[0])
         else:
             lst = style(x)
-    elif reduce(product, a.shape) == 0:
+    elif functools.reduce(product, a.shape) == 0:
```

Comment:
Not polluting the global namespace. Previously there was also `from functools import reduce`.

As for why that's there - `reduce` is no longer a global in python 3